### ISSUE TITLE
v0.5 research milestone: reproducible open-model evidence tier

### ISSUE THREAD

**Issue by hyouo** [hyouo/impact_of_guided_questioning_on_llm_bias#24]
## Goal

Move the repository from transparent L0/L1 demonstrations to a small but rigorous L2/L3 open-model evidence tier without weakening the existing learning-first structure.

## Scope

Use one small openly available causal language model first. Freeze the exact model revision, tokenizer revision, chat/template behavior, dependency versions, and paired prompt dataset before adding more models.

### O01 — Tokenization and format decomposition

For paired prompts, separately record:

- raw strings and token IDs;
- sequence length and final position;
- next-token logits and Jensen–Shannon divergence;
- per-layer final-position hidden-state cosine distance;
- matched controls for equal length and approximately matched semantics.

### O02 — Prefix feedback

Compare two externally supplied prefixes while holding the base prompt and weights fixed. Record full next-token distributions and multi-step trajectory divergence. Do not claim the prefixes had equal natural probability.

### O03 — Residual activation patching

Use clean/corrupted prompt pairs with matching tensor shape. Scan layer and position, measure target logit-difference recovery, and include:

- random-vector control;
- wrong-position control;
- wrong-layer control;
- clean-to-corrupt and corrupt-to-clean directions;
- multiple prompt templates.

### O04 — Steering specificity

Extend C12 to a real model with:

- target direction and sign reversal;
- dose response;
- at least 100 equal-norm random directions;
- orthogonalized controls;
- wrong layer and wrong position;
- non-target behavior and perplexity side-effect metrics.

## Evidence and reporting contract

Every run must preserve:

```text
model and tokenizer revisions
prompt pair IDs and token IDs
code commit
seed and deterministic settings
intervention site and strength
raw logits / hidden metrics
positive and negative controls
all skipped, failed, null, and falsifying cases
allowed conclusion and forbidden inference
```

Use statuses that distinguish `supported`, `falsified`, `inconclusive`, `skipped`, and `error`; do not map missing dependencies or execution failures to theory failure.

## Acceptance criteria

- [ ] A versioned paired-prompt dataset with licenses and provenance.
- [ ] A model manifest with immutable revisions and hashes where available.
- [ ] Offline/cache-first execution; no unreviewed `trust_remote_code`.
- [ ] Machine-readable JSON schema for open-model runs.
- [ ] Tests using a tiny local fixture or mocked model outputs.
- [ ] At least one complete CPU-runnable baseline.
- [ ] Every experiment linked to a course chapter and dedicated lab guide.
- [ ] CI validates schemas and fixtures without downloading large weights.
- [ ] A result report that retains null and falsifying examples.
- [ ] No operational jailbreak payloads or unsafe optimization loops.

## Non-goals

- claiming a universal mechanism from one model;
- training production-scale SAEs in this milestone;
- using an LLM judge as the sole metric;
- downloading mutable model revisions in CI;
- presenting a patching heatmap as complete source code.
